docs/dns-minikube.md

```diff
@@ -0,0 +1,170 @@
+        - --crd-source-kind=DNSEndpoint
+        - --registry=txt
+        - --provider=coredns
+        - --log-level=debug # debug only
+        env:
+        - name: ETCD_URLS
+          value: http://10.102.147.224:2379
+
+$ kubectl apply -f external-dns.yaml
+```
+## Enable DNS for federation resources
+### Installing metellb for Service
+Install metallb in each member cluster to make LoadBanlancer type Service work.
+```
+$ helm --kube-context cluster1 install --name metallb stable/metallb
+$ helm --kube-context cluster2 install --name metallb stable/metallb
+```
+Apply configmap to configure metallb in each cluster.
+```
+$ cat configmap-1.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metallb-config
+data:
+  config: |
+    peers:
+    - peer-address: 10.0.0.1
+      peer-asn: 64501
+      my-asn: 64500
+    address-pools:
+    - name: default
+      protocol: bgp
+      addresses:
+      - 192.168.20.0/24
+$ cat configmap-2.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metallb-config
+data:
+  config: |
+    peers:
+    - peer-address: 10.0.0.1
+      peer-asn: 64501
+      my-asn: 64500
+    address-pools:
+    - name: default
+      protocol: bgp
+      addresses:
+      - 172.168.20.0/24
+
+$ kubectl apply --context cluster1 -f configmap-1.yaml
+$ kubectl apply --context cluster2 -f configmap-2.yaml
+```
+### Creating service resources
+After metallb works, create sample deployment and service from [sample](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/ingressdns-with-externaldns.md). Make service as LoadBalancer type.
+```
+sed -i 's/NodePort/LoadBalancer/' example/sample1/federatedservice-template.yaml
+```
+
+Create ServiceDNSRecord to make DNS work for service.
+```
+$ cat <<EOF | kubectl create -f -
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: Domain
+metadata:
+  # Corresponds to <federation> in the resource records.
+  name: test-domain
+  # The namespace running federation-controller-manager.
+  namespace: federation-system
+# The domain/subdomain that is setup in your externl-dns provider.
+domain: example.org
+---
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: ServiceDNSRecord
+metadata:
+  # The name of the sample service.
+  name: test-service
+  # The namespace of the sample deployment/service.
+  namespace: test-namespace
+spec:
+  # The name of the corresponding `Domain`.
+  domainRef: test-domain
+  recordTTL: 300
+EOF
+```
+### Enable the ingress controller
+```
+$ minikube -p cluster1 addons enable ingress
+$ minikube -p cluster2 addons enable ingress
+```
+
+After ingress controller enabled, create sample deployment, service and ingress from [sample](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/ingressdns-with-externaldns.md). Make ingress with "example.org" for this tutorial example.
+```
+sed -i 's/example.com/example.org/' example/sample1/federatedingress-template.yaml
```

Comment:
Instead of replacing `example.com` to `example.org`, can you use `example.com` for the tutorial? This can keep consistent with GKE document as well.